Added support and integration with 'ammonext' physics.

alkavan opened the pull request:
It seems ammo physics is not so easy to integrate correctly, what do you think about physics branch?

* Added example Sphere component PhysicsSphere
* Added example Plane component PhysicsPlane

sasha240100 commented on app/components/PhysicsPlane.js:
```diff
@@ -0,0 +1,42 @@
+
+import {MeshComponent} from '@whs/core/MeshComponent';
+
+export class PhysicsPlane extends Plane {
```
Is there a need for `PhysicsPlane`, `PhysicsSphere` ... ?
I mean that they are used only **once** in code.

sasha240100 replied:
`BasicComponent` was created to show how to organize a first simple component (similarly to React).

sasha240100 commented on app/components/PhysicsPlane.js:
```diff
@@ -0,0 +1,42 @@
+  constructor(params = {}) {
+
+    params.modules = [
+      new PlaneModule({
```
That will overwrite all `modules` array. **I'd recommend** using `this.applyModule()` after calling `super(...)`.

```javascript
this.applyModule(
  new PlaneModule({
    mass: 0
  })
);
```

sasha240100 replied:
But, anyway, that's just my suggestion of how to do it properly. For very specific component you can ignore that. But if you want to make your components universal - use the code above.

sasha240100 commented on webpack.config.babel.js:
```diff
@@ -28,6 +28,10 @@ const config = {
         test: /\.(glsl|frag|vert)$/, loader: 'glslify-loader', exclude: /node_modules/
+      },
+      {
+        test: /\.wasm$/,
```
It seems we don't use `wasm` in this example. If you want to use `wasm` build - [read this](https://gist.github.com/sasha240100/1e0c0894b2dd98623ef68f89d4ef87ab) and get those files. 
You need `ammoloader.js` and `ammo.wasm` files.

```javascript
new WorldModule({
   wasmBuffer: 'ammo.wasm',
   ammo: 'ammoloader.js'
})
```

See [this code](https://github.com/WhitestormJS/physics-module-ammonext/blob/master/src/worker.js#L313).

alkavan commented:
I've made a commit being more explicit about everything, better for example.
However I did not realize the Ammo.js integration (with THREE) is still in this state.

I saw your repository at https://github.com/sasha240100/ammojs-build-files
I see now how I missed some parts. My whole point was to use the WebAssembly example.
Will do another commit with proper wasm integration using your builds on that repo.

sasha240100 commented:
@alkavan There is no need for wasm (no extra optimization on loading time yet)

alkavan commented:
@sasha240100 I've updated the PR. Removed wasm dependency, and Improved the demo.
I'm gonna try loading the wasm on my project first, I see it's still too experimental perhaps.
